### PR TITLE
Add integration tests for full file contents and metadata

### LINKED ISSUE #241: Follow-up about content type
From https://github.com/NebulousLabs/skynet-js/pull/239#discussion_r545863303:

> What happens if we don't specify the content type? Would it be detected?
No need to cover this now but it might be interesting to verify the behaviour.

### PR DESCRIPTION
**Pull request by mrcnski**
**Note:** Tests currently failing due to access controls on siasky.net blocking the skynet-skylink header.

Closes #241

### CODE REVIEW

**Review by ro-tex — DISMISSED** [review]

File: src/integration.test.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { MAX_GET_ENTRY_TIMEOUT } from "./registry";
 import { MAX_REVISION } from "./utils";
 
 // TODO: Use siasky.dev when available.
```
Comment (ro-tex) [inline src/integration.test.ts]: This line can go now. ;)

File: src/integration.test.ts [hunk 2]
```diff
@@ -150,15 +151,30 @@ describe("Upload and download end-to-end tests", () => {
...
+    const { skylink } = await client.uploadFile(file);
+    expect(skylink).not.toEqual("");
+
+    // Get file content and check returned values.
+
+    const { data, contentType } = await client.getFileContent(skylink);
 
     expect(data).toEqual(expect.any(Object));
     expect(data).toEqual(json);
+    expect(contentType).toEqual("application/octet-stream");
```
Comment (ro-tex) [inline src/integration.test.ts]: What happens if we specify the file's name and it has an extension? The content type should be guessed correctly, right? Can we add a test for that as well?

Actually, if the content allows it, we should get some content type heuristics even without a filename. All that "will read up to X bytes and guess the content" magic. Doesn't that work or is this JSON too short?

**Review by ro-tex — DISMISSED** [review]
LGTM. I left some comments but nothing is blocking.

File: src/integration.test.ts (hunk 1 above)
Comment (mrcnski) [reply src/integration.test.ts]: Looks like siasky.dev is still not working (the registry is broken), going back to siasky.net for now

File: src/integration.test.ts (hunk 2 above)
Comment (mrcnski) [reply src/integration.test.ts]: I don't know why it doesn't infer from the content, but I added a test case for inferring from filename

**Review by ro-tex — DISMISSED** [review]

**Review by dghelm — APPROVED** [review]

**Review by ro-tex — APPROVED** [review]

**Pull request merged**